From 81e6bce2ae2fc2ba226e0004c14f7aa1c75bc219 Mon Sep 17 00:00:00 2001
From: Sven Neumann <neumann@teufel.de>
Date: Mon, 24 Feb 2014 10:49:09 +0100
Subject: [PATCH] Util: fix rectangle sizes returned by
 dfb_updates_get_rectangles()

The rectangles used to be 1 pixel too small in both dimensions.
---
 src/misc/util.c | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/misc/util.c b/src/misc/util.c
index 9600135..a1e540f 100644
--- a/src/misc/util.c
+++ b/src/misc/util.c
@@ -418,8 +418,8 @@ dfb_updates_get_rectangles( DFBUpdates   *updates,
                     for (n=0; n<updates->num_regions; n++) {
                          ret_rects[n].x = updates->regions[n].x1;
                          ret_rects[n].y = updates->regions[n].y1;
-                         ret_rects[n].w = updates->regions[n].x2 - updates->regions[n].x1;
-                         ret_rects[n].h = updates->regions[n].y2 - updates->regions[n].y1;
+                         ret_rects[n].w = updates->regions[n].x2 - updates->regions[n].x1 + 1;
+                         ret_rects[n].h = updates->regions[n].y2 - updates->regions[n].y1 + 1;
                     }
 
                     break;
@@ -432,8 +432,8 @@ dfb_updates_get_rectangles( DFBUpdates   *updates,
 
                ret_rects[0].x = updates->bounding.x1;
                ret_rects[0].y = updates->bounding.y1;
-               ret_rects[0].w = updates->bounding.x2 - updates->bounding.x1;
-               ret_rects[0].h = updates->bounding.y2 - updates->bounding.y1;
+               ret_rects[0].w = updates->bounding.x2 - updates->bounding.x1 + 1;
+               ret_rects[0].h = updates->bounding.y2 - updates->bounding.y1 + 1;
                break;
      }
 }
-- 
1.8.3.2

